automatically create output directory if it doesn't exist

```
regier@deeplearning-01:~/bliss$ poetry run bliss +experiment=sdss_sleep
Error executing job with overrides: ['+experiment=sdss_sleep']
Traceback (most recent call last):
  File "/home/regier/bliss/bliss/main.py", line 16, in main
    task(cfg)
  File "/home/regier/bliss/bliss/train.py", line 60, in train
    assert Path(p).exists(), f"path {Path(p).as_posix()} does not exist"
AssertionError: path /home/regier/bliss/output does not exist

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
```

--- a/bliss/train.py
+++ b/bliss/train.py
@@ -60,7 +60,7 @@ def train(cfg: DictConfig):
         path = Path(paths[key])
         if key == "output" and not path.exists():
             path.mkdir(parents=True)
-        assert path.exists(), f"path {path.as_posix()} does not exist"
+        assert path.exists(), f"path for {key} ({path.as_posix()}) does not exist"
     setup_seed(cfg)
 
     # setup dataset.

--- a/bliss/train.py
+++ b/bliss/train.py
@@ -58,9 +58,11 @@ def train(cfg: DictConfig):
     paths = OmegaConf.to_container(cfg.paths, resolve=True)
     for key in paths.keys():
         path = Path(paths[key])
-        if key == "output" and not path.exists():
-            path.mkdir(parents=True)
-        assert path.exists(), f"path for {key} ({path.as_posix()}) does not exist"
+        if not path.exists():
+            if key == "output":
+                path.mkdir(parents=True)
+            else:
+                raise FileNotFoundError(f"path for {key} ({path.as_posix()}) does not exist")
     setup_seed(cfg)
 
     # setup dataset.

--- a/bliss/train.py
+++ b/bliss/train.py
@@ -56,8 +56,11 @@ def train(cfg: DictConfig):
 
     # setup paths and seed
     paths = OmegaConf.to_container(cfg.paths, resolve=True)
-    for p in paths.values():
-        assert Path(p).exists(), f"path {Path(p).as_posix()} does not exist"
+    for key in paths.keys():
+        path = Path(paths[key])
+        if key == "output" and not path.exists():
+            path.mkdir(parents=True)
+        assert path.exists(), f"path {path.as_posix()} does not exist"
     setup_seed(cfg)
 
     # setup dataset.

